packages/medusa/src/services/product.ts

```diff
@@ -0,0 +1,1067 @@
+  protected readonly imageRepository_: typeof ImageRepository
+  protected readonly cartRepository_: typeof CartRepository
+  protected readonly productVariantService_: ProductVariantService
+  protected readonly productCollectionService_: ProductCollectionService
+  protected readonly searchService_: SearchService
+  protected readonly eventBus_: EventBusService
+  protected readonly priceSelectionStrategy_: PriceSelectionStrategy
+
+  static readonly IndexName = `products`
+  static readonly Events = {
+    UPDATED: "product.updated",
+    CREATED: "product.created",
+    DELETED: "product.deleted",
+  }
+
+  constructor({
+    manager,
+    productRepository,
+    productVariantRepository,
+    productOptionRepository,
+    eventBusService,
+    productVariantService,
+    productCollectionService,
+    productTypeRepository,
+    productTagRepository,
+    imageRepository,
+    searchService,
+    cartRepository,
+    priceSelectionStrategy,
+  }: InjectedDependencies) {
+    super({
+      manager,
+      productRepository,
+      productVariantRepository,
+      productOptionRepository,
+      eventBusService,
+      productVariantService,
+      productCollectionService,
+      productTypeRepository,
+      productTagRepository,
+      imageRepository,
+      searchService,
+      cartRepository,
+      priceSelectionStrategy,
+    })
+
+    this.manager_ = manager
+
+    this.productOptionRepository_ = productOptionRepository
+
+    this.productRepository_ = productRepository
+
+    this.productVariantRepository_ = productVariantRepository
+
+    this.eventBus_ = eventBusService
+
+    this.productVariantService_ = productVariantService
+
+    this.productCollectionService_ = productCollectionService
+
+    this.productTypeRepository_ = productTypeRepository
+
+    this.productTagRepository_ = productTagRepository
+
+    this.imageRepository_ = imageRepository
+
+    this.searchService_ = searchService
+
+    this.cartRepository_ = cartRepository
+
+    this.priceSelectionStrategy_ = priceSelectionStrategy
+  }
+
+  /**
+   * Lists products based on the provided parameters.
+   * @param selector - an object that defines rules to filter products
+   *   by
+   * @param config - object that defines the scope for what should be
+   *   returned
+   * @return the result of the find operation
+   */
+  async list(
+    selector: FilterableProductProps | Selector<Product> = {},
+    config: FindProductConfig = {
+      relations: [],
+      skip: 0,
+      take: 20,
+      include_discount_prices: false,
+    }
+  ): Promise<Product[]> {
+    return await this.atomicPhase_(async (manager) => {
+      const productRepo = manager.getCustomRepository(this.productRepository_)
+
+      const { q, query, relations } = this.prepareListQuery_(selector, config)
+
+      const [filteredRelations, shouldLoadStrategyPrices] =
+        this.omitPriceRelationIfExists_(relations)
+
+      if (q) {
```

Comment:
question: Does that means that if you search with a free text search, you don't get the prices included even if you ask to have them?

Reply:
I don't think so, it means that you defer the loading of price until they are set through `setAdditionalPrices()` https://github.com/medusajs/medusa/blob/d8c870ef0f5b6d767f2f1bdf54a43828c958ae91/packages/medusa/src/services/product.ts#L181 Does that answer your question?

Reply:
no, you are returning in the if statement from what I can see, so you never reach that ternary expression

Reply:
ah yes totally sorry! It was initially fixed in a different PR, but I think I didn't resolve the conflict properly when I merged. Good catch!!

Reply:
Ahah no problem, in that case, can I let you check that other conflicts have not generate some other unexpected behaviour just to be sure ☺️